s3tests_boto3/functional/test_iam.py

```diff
@@ -23,6 +23,7 @@
 @attr(assertion='succeeds')
 @attr('user-policy')
 @attr('test_of_iam')
+@attr('fails_on_dbstore')
```

Comment:
do these tests fail against dbstore even if they're configured correctly? if not, then i don't think `fails_on_dbstore` is the right way to address this

Reply:
sure...I haven't tested any sts/iam cases on dbstore..this change was more like an interim fix as I was planning to address s3test failures on dbstore soon. 

Will check and update. Thanks!

Reply:
@soumyakoduri I'm going to revert the commit(s) from the iam policy tests from ceph-master. Once we figure do some tests with dbstore on sts/iam then maybe we can merge this or see what work still needs to be done.

Reply:
The 'iam section' was introduced as part of sts tests, which was merged long back and not as part of #453. @TRYTOBE8TME  : any idea what might be happening here?

Reply:
In https://github.com/ceph/ceph/pull/46660, for running s3tests in teuthology, iam tests/ststests are filtered out by adding !test_of_iam, !test_of_sts here: https://github.com/ceph/ceph/blob/main/qa/tasks/s3tests.py#L456, 
Can !test_of_iam, !test_of_sts be added to qa/suites/rgw/dbstore/tasks/rgw_s3tests.yaml, to filter these tests out.

Reply:
> @soumyakoduri I'm going to revert the commit(s) from the iam policy tests from ceph-master. Once we figure do some tests with dbstore on sts/iam then maybe we can merge this or see what work still needs to be done.

these tests are failing on dbstore even with iam section include. So I think we can merge this commit for now.

Reply:
> In [ceph/ceph#46660](https://github.com/ceph/ceph/pull/46660), for running s3tests in teuthology, iam tests/ststests are filtered out by adding !test_of_iam, !test_of_sts here: https://github.com/ceph/ceph/blob/main/qa/tasks/s3tests.py#L456, Can !test_of_iam, !test_of_sts be added to qa/suites/rgw/dbstore/tasks/rgw_s3tests.yaml, to filter these tests out.

thanks Pritha!  Though it will work for teuthology run, it might help to ideally skip these tests if the details are missing in config file , in case they are run locally. For now I am keeping the tracker issue - https://tracker.ceph.com/issues/56421 open with lower priority.

Reply:
@soumyakoduri : what command did you use to run the tests locally?

Reply:
If filtering out the tests is the thing what we want then, where we set the parameters in s3tests.py we can have a new parameter named `iam_present_or_not` and based on IAM section is present or not we can apply the extra attributes based on what we wanna execute maybe?

Reply:
Results with this commit - http://pulpito.front.sepia.ceph.com/soumyakoduri-2022-06-30_08:23:39-rgw:dbstore-wip-skoduri-dbstore-lc-distro-default-smithi/

Reply:
@soumyakoduri when I ran the iam tests on db_store with vstart I only got 7 test failures/errors. A majority of these tests failed because the RGW returned a `404` instead of a `403`. Here is a log with the output: https://paste.sh/KZarwzOE#z7mNOTuIXKAw3QndMf9i3sA5

Is this what you are seeing locally as well?

Reply:
I guess so .. though now on my current WIP branch (https://github.com/soumyakoduri/ceph/tree/wip-skoduri-dbstore-rm), I see very few failures  - https://paste.sh/58ZJ495g#9Qx0tGMTFaXSHp7Hucrpd9S0